squad/core/models.py

```diff
@@ -374,7 +358,28 @@ def overall(self):
         return self.filter(suite=None)
 
 
-class Status(models.Model):
+class TestSummaryBase(object):
+    @property
+    def tests_total(self):
+        return self.tests_pass + self.tests_fail + self.tests_missing
+
+    @property
+    def pass_percentage(self):
+        if self.tests_pass > 0:
+            return 100 * (float(self.tests_pass) / float(self.tests_total))
+        else:
+            return 0
+
+    @property
+    def non_pass_percentage(self):
+        return 100 - self.pass_percentage
+
+    @property
+    def has_tests(self):
+        return (self.tests_pass + self.tests_fail) > 0
```

Comment:
hmm, maybe 'missing' should also be included here? In general I don't think that 'missing' is a proper name. 'skip' is a valid result and we should support it at some point.